nemo/collections/multimodal/speech_llm/models/modular_s2s_models.py

```diff
@@ -1139,27 +1158,32 @@ def resample(audio, audio_lens, orig_sample_rate, target_sample_rate):
                 )
                 sliced_text_channel = text_channel[: answer_codec.shape[0]]
 
-            if getattr(self.cfg, 'speech_delay', False):
```

Comment:
why removing speech_delay? can't they be combined? predict_source_text + speech_delay?

Reply:
I am not removing - it is a diff problem.

Reply:
can we support predict_source_text + speech_delay?

Reply:
Of course - added a TODO. Should be easy. I will implement later in a more comprehensive way with cascaded delays in all channels.